src/org/apache/commons/httpclient/HttpMethodDirector.java

```diff
@@ -507,14 +507,7 @@ private void executeWithRetry(final HttpMethod method)
                     LOG.info("Retrying request");
                 }
             }
-        } catch (IOException e) {
-            if (this.conn.isOpen()) {
-                LOG.debug("Closing the connection.");
-                this.conn.close();
-            }
-            releaseConnection = true;
-            throw e;
-        } catch (RuntimeException e) {
+        } catch(IOException | RuntimeException e) {
```

Comment:
Please, revert this change, this is a forked class, better keep the changes to a minimum and essential ones.